ddclient

```diff
@@ -2063,10 +2063,11 @@ sub geturl {
     $headers .= "\n" if $headers ne '' && substr($headers, -1) ne "\n";
     $request .= $headers;
     $request .= "\n";
+    # RFC 7230 says that all lines before the body must end with <cr><lf>.
+    (my $rq = $request) =~ s/(?<!\r)\n/\r\n/g;
```

Comment:
Why do we need ``$rq`` exactly? It is only used at one other place and I think it is redundant.

Reply:
`$rq` is what is actually sent to the server. `$request` is only used for log messages.

Reply:
Mostly I didn't want to change too much in one commit, though including `\r` in syslog, stdout, and/or emailed logs also seems risky.